src/serial_model.c

```diff
@@ -126,19 +126,25 @@ static int serial_read(void)
   int           xpt_read_ret =
       xpt.vtbl->read(&xpt, buf+HEADER_LENGTH, BUFLEN-HEADER_LENGTH, &addr, &port);
 
-  if (xpt_read_ret < 0)
+  if (xpt_read_ret >= 0)
+    {
+      tm     = time_in_milliseconds();
+      buflen = xpt_read_ret + HEADER_LENGTH;
+      enqueuer_stats_record_datagram(&est, buflen);
+      header_add(buf, xpt_read_ret, tm, addr, port);
+      ++count;
+      return 0;
+    }
+  else if (xpt_read_ret == -2)
```

Comment:
I was wondering where the -2 came from and ran on a long sequence of files before I reached it in xport_udp.c ("recvfrom_ret = -2; /* special return on interruption */").  Could we give that some name?  I had been wondering if it were intended to be related to EINTR or a signal or something.  A name would be easily searchable and hopefully clearer.